### PR TITLE
Topic6

### PR DESCRIPTION
**Pull request by stringie**

### CODE REVIEW

File: topic6/contracts/PeopleBalances.sol [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    
+    uint256 public crowdsaleEndTime = now + 5 minutes;
+    uint256 public ownerWithdrawStartTime = now + 365 days;
+    uint16 constant public EXCHANGE_RATE = 5;
+
+    mapping (address=>uint256) public balances;
+    mapping (address=>bool) public hasHeldTokens;
+    address[] public tokenHolders;
+
+    function buy() public payable isCrowdsale {
```
Comment (ArnaudovSt) [inline topic6/contracts/PeopleBalances.sol]: Where is that `isCrowdsale` modifier?

File: topic6/contracts/PeopleBalances.sol [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        
+        balances[msg.sender] += msg.value * EXCHANGE_RATE;
+    }
+    
+    function transferTokens(address _to, uint256 _amount) public {
+        require(crowdsaleEndTime < now);
+        require(balances[msg.sender] >= _amount);
+        require(_to != address(0));
+
+        if (!hasHeldTokens[_to]) {
```
Comment (ArnaudovSt) [inline topic6/contracts/PeopleBalances.sol]: DRY principle -> extract internal/private function?

File: topic6/contracts/PeopleBalances.sol [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            hasHeldTokens[_to] = true;
+            tokenHolders.push(_to);
+        }
+
+        balances[msg.sender] -= _amount;
+        balances[_to] += _amount;
```
Comment (ArnaudovSt) [inline topic6/contracts/PeopleBalances.sol]: Overflow chance? Use SafeMath library - [link](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/math/SafeMath.sol)

**Review by ArnaudovSt — CHANGES_REQUESTED** [review]

File: topic6/contracts/PeopleBalances.sol (hunk 1 above)
Comment (stringie) [reply topic6/contracts/PeopleBalances.sol]: Oops.

**Pull request merged**